### DISCUSSION TITLE
I need help with advance setup of IQueryable with NSubstitute. An example will help.

### DISCUSSION THREAD

**Discussion by vaishnavravi33** [ramantsitou/MockQueryable#76] [Q&A] [answered]
I found advance setup with Moq but not found with NSubstitute.

Moq advance setup example - 

```
//Aditional setup for FindAsync
mock.Setup(x => x.FindAsync(userId)).ReturnsAsync((object[] ids) =>
{
    var id = (Guid)ids[0];
    return users.FirstOrDefault(x => x.Id == id);
});
var userRepository = new TestDbSetRepository(mock.Object);

//Execution FindAsync
var user = await ((DbSet<UserEntity>) userRepository.GetQueryable()).FindAsync(userId);
```

Any one have example to setup the same with NSubstitute please share. Thank you !!

**Comment by ramantsitou** [comment, marked as answer]
Hello @vaishnavravi33. There is indeed no an example for `FindAsync` there are [few examples](https://github.com/romantitov/MockQueryable/blob/44663dc1e299cd036d0ae832300dbf59fc9856b2/src/MockQueryable/MockQueryable.Sample/MyServiceNSubstituteTests.cs#L152) for `AddAsync`.  Hope it can help you.
